ui/hooks/gasFeeInput/useGasFeeInputs.js

```diff
@@ -136,6 +136,10 @@ export function useGasFeeInputs(
     Number(hexToDecimal(transaction?.txParams?.gas ?? '0x0')),
   );
 
+  const properGasLimit = Number(
```

Comment:
What about this case:

1. Gas limit is estimated to 100,000
2. User opens the gas popover, edits the gas limit to 200,000 and saves the changes
3. `transaction?.txParams?.gas` is now 200,000
4. User opens the gas popover again, and edits the gas limit to 150,000
5. `properGasLimit` will be 200,000 and `gasLimit` will be 150,000 and the user will see a warning, even though 150,000 is higher than the original estimate of 100,000

I think we need to save the original estimate on the transaction metadata.

To do that, the following code in `app/scripts/controllers/transactions/index.js` should be edited

```
    if (defaultGasLimit && !txMeta.txParams.gas) {
      txMeta.txParams.gas = defaultGasLimit;
    }
```

It could be changed to 

```
  if (defaultGasLimit && !txMeta.txParams.gas) {
      txMeta.txParams.gas = defaultGasLimit;
      txMeta.originalGasEstimate = defaultGasLimit;
  }
```

Reply:
@danjm I added `originalGasEstimate` property into `txMeta` and now it is work correctly.
You can see changes here: 8e8a6ecf4a162029c09ca4616d393bde27dc713a
Is it okay now?